source/storing-data/provenance.rst

```diff
@@ -2,3 +2,213 @@ Decentralized Provenance Tracking
 =================================
 .. contents::
    :local:
+
+QIIME 2 provides automatic, integrated, and decentralized tracking of analysis
+metadata, including information about the host system, the computing
+environment, Actions performed, parameters passed, primary sources cited, and
+more. We describe all of this information about "how an analysis was
+performed" as "provenance".
+
+The notion of a QIIME 2 :term:`Result` is central here. Whenever an
+:term:`Action` is performed on some data with QIIME 2, the framework
+captures relevant metadata about the action and environment and stores it in
+the Action's Result. When/if that Result is saved as an :term:`Archive`, the
+captured provenance data is stored within the Archive as well.
+:ref:`provenance-structure` contains a detailed discussion of the
+file structure which holds provenance metadata.
+
+.. note::
+   *When* Results are saved as Archives is interface-defined.
+   Results are saved automatically by ``q2cli``, every time a user runs a command.
```

Comment:
I think about this a little bit differently. The archive is the name for our structure, a result is basically always backed by an archive (although I could see parallelism making this not always true if you had a "future result"). What this note is really demonstrating is whether or not the archive becomes a QZA/QZV (zipped archive).

Reply:
OK, this is a really important distinction for me. The glossary defines Archives in terms of a directory structure, but it sounds like that structure may exist in main memory rather than on disk?

![image](https://user-images.githubusercontent.com/39198770/126401658-9dbf27d4-d828-409d-a2c6-a7ee64f03452.png)

I'm going to take a crack at rewriting the relevant sections with this distinction in mind.

Reply:
No the glossary is correct here, the distinction I am making is that results are always saved as archives, whether that archive then becomes a qza is a different matter (and up to the interface)

Reply:
Thanks! A couple last questions.

- It sounds like the Archive's directory structure is stored on disk somewhere, but generally not in a place the user cares about (e.g in `TEMPDIR`)?
- For practical purposes, is the distinction between an Archive and a .qza primarily about about whether it has been `.save()`d? (`_ZipArchive.save()` just writes source files to a zipfile at `destination`, so assuming a `_ZipArchive`, about zipped-ness and file location?)

Tangentially, if a .qza is not technically an Archive, we don't really have an "official" word for .qza and .qzv files, do we? This isn't a bad thing, but I suspect I'm not the only person the distinction has been lost on.

Reply:
That is correct on all points, and the lack of a word for qza/qzv is somewhat annoying.